pkg/controller/v1alpha2/experiment/validator/validator.go

```diff
@@ -140,8 +141,13 @@ func (g *DefaultValidator) validateSupportedJob(job *unstructured.Unstructured)
 }
 
 func (g *DefaultValidator) validateForCreate(inst *experimentsv1alpha2.Experiment) error {
+	isErrNoRows := func (e error) bool {
```

Comment:
Why should we use such a func instead of sql.ErrNoRows, is there any problem that we want to avoid?

Reply:
sql.ErrNoRows is raised by db interface in grpc server, in grpc client side,  we cannot get sql.ErrNoRows, which has been wrapped.

Reply:
It looks hacky. Just wondering if there is a better way to handle this

Reply:
I feel that DB errors should not be exposed to controller

Reply:
Could we catch the error in GetExperimentFromDB? If the error is  sql.ErrNoRows we return nil, or we return error to the controller.

Reply:
In fact, I don't like handling like this. I will try to investigate what general solution of gprc error is
thanks for your comments

Reply:
Per investigation, we have to parse the Error message of GetExperimentFromDB (in grpc client side) to decide if grpc server GetExperiment error is sql.ErrNoRows or not owning to grpc implementation.
Considering that in future release, if we support more db backend, GetExperiment may not return sql.ErrNoRows, it can be modeldb.NoRecords and so on. So I prefer to define a general new api as below (since here call GetExperiment to check if a same name experiment exists in DB).
WDYT?
```
rpc PreCheckRegisterExperiment(RegisterExperimentRequest) returns (PreCheckRegisterExperimentReply, error)

message PreCheckRegisterExperimentReply {
    bool can_register = 1;
}
```

Reply:
how will you handle similar errors for other api calls?  eg: Alreadyexists for RegisterExperiment etc

Reply:
For others APIs, our use cases don't need care what the error type is now when error is not nil.

Reply:
Can we update managerclient.go as follows:

```go
func (d *DefaultClient) GetExperimentFromDB(instance *experimentsv1alpha2.Experiment) (*api_pb.GetExperimentReply, error) {
	request := &api_pb.GetExperimentRequest{
		ExperimentName: instance.Name,
	}
	if exp, err := commonv1alpha2.GetExperiment(request); err != nil {
		if !isErrNoRows(err) {
			return nil, fmt.Errorf("Fail to check record for the experiment in DB: %v", err)
		}
	}
	return exp, nil
}
```

Then the error will be handled in manager client and not be exposed to the controller.

Reply:
I had used "PreCheckRegisterExperiment" to replace "GetExperimentFromDB" in latest commit. I think we should avoid to parse Error message string to judge what its original error in grpc server side is, especially when we support more DB backends, we must refactor `isErrNoRows` to parse the error message string to match other backends ErrNoRows counterparts.

Reply:
SGTM